File: baekjoon/baekjoon_11726.java
// https://www.acmicpc.net/problem/11726
// 백준 11726번 : 2*n 타일링

import java.util.Scanner;

public class baekjoon_11726 {

    static int[] way;

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        way = new int[n + 1];
        way[0] = way[1] = 1;
        for (int i = 2; i <= n; i++) {
            way[i] = (way[i - 1] + way[i - 2]) % 10007;
        }

        System.out.println(way[n]);
    }
}

File: baekjoon/baekjoon_11726.py
# https://www.acmicpc.net/problem/11726
# 백준 11726 : 2xn 타일링
# LEVEL : Silver
import sys


def solution():
    N = int(sys.stdin.readline())

    memo = [0] * (N+1)
    memo[0] = 1
    memo[1] = 1

    for i in range(2, N+1):
        memo[i] = memo[i-1] + memo[i-2]

    print(memo[N] % 10007)


solution()
